### PR TITLE
remove reinterpret_cast in rhs.H

### PR DESCRIPTION
**Pull request by zingale**

### CODE REVIEW

**Comment by zingale** [comment]
tests: http://groot.astro.sunysb.edu/Microphysics/test-suite/gfortran/2024-01-02-003/index.html

**Comment by zingale** [comment]
I think that part of the issue here is that `is_jacobian_term_used<>()` checks if a term is Jacobian term is used by comparing to the 0-based species, so when we change the loops to start at 1 instead of 0, we need to shift the indexing used in `is_jacobian_term_used`

**Comment by zingale** [comment]
all tests pass with this change:
http://groot.astro.sunysb.edu/Microphysics/test-suite/gfortran/2024-03-09-002/index.html

File: networks/rhs.H [hunk 1]
```diff
@@ -273,19 +273,17 @@ constexpr int is_jacobian_term_used ()
 }
 
 AMREX_GPU_HOST_DEVICE AMREX_INLINE
-void dgesl (RArray2D& a1, RArray1D& b1)
+void dgesl (RArray2D& a, RArray1D& b)
```
Comment (yut23) [inline networks/rhs.H:276]: `a` can be made const here

**Review by yut23 — APPROVED** [review]
I went through and did the index shifts for myself, and got the same code.

File: networks/rhs.H (hunk 1 above)
Comment (zingale) [reply networks/rhs.H]: updated

**Pull request merged**